### PR TITLE
[Synthetics] Document the support matrix for the various Synthetics components

### PR DESCRIPTION
**Pull request by paulb-elastic**
Synthetics is made up of a number of product components and is run differently to some of the other Elastic solutions, and so does not really fit in with the main [Support matrix](https://www.elastic.co/support/matrix).

This PR is intended to create a support matrix page specifically for Synthetics (similar to something similar that [APM Agent](https://www.elastic.co/guide/en/apm/guide/current/agent-server-compatibility.html) has).

### CODE REVIEW

**Review by andrewvc — APPROVED** [review]
LGTM, I'm OK filling in the recorder bit later

File: docs/en/observability/synthetics-support-matrix.asciidoc [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+** GA as defined in the standard https://www.elastic.co/support/matrix[Support matrix]
+** For running lightweight monitors via YML configuration running on self-managed infrastructure
+*** Browser based monitors are not supported in this configuration
+* Standalone {agent}
+** GA as defined in the standard https://www.elastic.co/support/matrix[Support matrix]
+** For running lightweight monitors via YML configuration running on self-managed infrastructure with standalone {agent}
+*** Browser based monitors are not supported in this configuration
+** Results for monitors configured in this way are be available in Uptime (not the {synthetics-app})
+* Synthetics Recorder
+** TBC
```
Comment (andrewvc) [inline docs/en/observability/synthetics-support-matrix.asciidoc:32]:
```suggestion
** Beta
```

Comment (paulb-elastic) [reply docs/en/observability/synthetics-support-matrix.asciidoc]: Added details as per a slack conversation

**Comment by paulb-elastic** [comment]
@colleenmcginnis moved from draft to ready for review if you get a chance

**Comment by colleenmcginnis** [comment]
@paulb-elastic what do you think about formatting this as a table? I think it would be easier for the reader to parse. If you're in favor, I can push these changes and we can discuss specific language (for example I'm not sure "Feature" is the right heading for the first column).

localhost_8000_guide_synthetics-support-matrix html

**Comment by paulb-elastic** [comment]
@colleenmcginnis my goodness, thank you so much - that is **so** much better 🙏 
Yes, please go ahead and merge the styling in if you wouldn't mind.
For `Feature`, how about something like `Product area`?

**Comment by paulb-elastic** [comment]
Thanks again @colleenmcginnis, merging

**Pull request merged**